src/main/java/org/ambraproject/rhino/service/impl/ArticleCrudServiceImpl.java

```diff
@@ -526,40 +526,93 @@ public Article readArticle(ArticleIdentifier articleIdentifier) {
 
   @Override
   public Collection<ArticleRevision> getArticlesPublishedOn(LocalDate fromDate, LocalDate toDate) {
+    return getArticlesPublishedOn(fromDate, toDate, null);
+  }
+
+  @Override
+  public Collection<ArticleRevision> getArticlesPublishedOn(LocalDate fromDate, LocalDate toDate, String bucketName) {
     return hibernateTemplate.execute(session -> {
-      Query query = session.createQuery("" +
-          "SELECT ar " +
+      String queryString = bucketName == null
```

Comment:
Would using an `if` be more readable, since the ternary statement is spanning multiple lines?
```
      final String queryString;
      if (bucketName == null) {
        queryString = "SELECT ar " +
            "FROM ArticleRevision ar " +
            "INNER JOIN ar.ingestion ai " +
            "INNER JOIN  ar.ingestion.article at " +
            "WHERE ai.publicationDate >= :fromDate AND ai.publicationDate <= :toDate " +
            "AND ar.revisionId IS NOT NULL";
      } else {
        queryString = "SELECT ar " +
            "FROM ArticleRevision ar, ArticleFile file " +
            "INNER JOIN ar.ingestion ai " +
            "INNER JOIN ar.ingestion.article at " +
            "WHERE ai.publicationDate >= :fromDate AND ai.publicationDate <= :toDate " +
            "AND ar.revisionId IS NOT NULL " +
            "AND file.ingestion = ai " +
            "AND file.bucketName = :bucketName";
      }
```

Reply:
Optionally, you can utilize `com.google.common.base.Joiner` to concatenate the strings:

```
private static final Joiner SPACE_JOINER = Joiner.on(' ');
```

```
      final String queryString;
      if (bucketName == null) {
        queryString = SPACE_JOINER.join("SELECT ar",
            "FROM ArticleRevision ar",
            "INNER JOIN ar.ingestion ai",
            "INNER JOIN  ar.ingestion.article at",
            "WHERE ai.publicationDate >= :fromDate AND ai.publicationDate <= :toDate",
            "AND ar.revisionId IS NOT NULL");
      } else {
        queryString = SPACE_JOINER.join("SELECT ar",
            "FROM ArticleRevision ar, ArticleFile file",
            "INNER JOIN ar.ingestion ai",
            "INNER JOIN ar.ingestion.article at",
            "WHERE ai.publicationDate >= :fromDate AND ai.publicationDate <= :toDate",
            "AND ar.revisionId IS NOT NULL",
            "AND file.ingestion = ai",
            "AND file.bucketName = :bucketName");
      }
```

Reply:
This code is in master and I merged master so seems like new code related to this ticket. I will take care of this in RARO-837